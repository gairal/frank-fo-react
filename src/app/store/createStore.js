import { applyMiddleware, compose, createStore } from 'redux';
import thunk from 'redux-thunk';
import { browserHistory } from 'react-router';
import reducers from './reducers';
import { updateLocation } from './location';

export default (initialState = {}) => {
  // ======================================================
  // Middleware Configuration
  // ======================================================
  const middleware = [thunk];

  // ======================================================
  // Store Instantiation and HMR Setup
  // ======================================================
  const store = createStore(
    reducers(),
    initialState,
    compose(applyMiddleware(...middleware)),
  );
  store.asyncReducers = {};

  // To unsubscribe, invoke `store.unsubscribeHistory()` anytime
  store.unsubscribeHistory = browserHistory.listen(updateLocation(store));

  return store;
};
