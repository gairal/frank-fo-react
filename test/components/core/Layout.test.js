/* eslint-disable function-paren-newline,comma-dangle  */
import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import createStore from '@/store/createStore';
import Routes from '@/routes';
import LayoutWith, { Layout as DumbLayout } from '@/components/core/Layout';

describe('components > core > Layout', () => {
  const store = createStore();
  const routes = new Routes(store);

  const component = renderer.create(
    <Provider store={store}>
      <Router>
        <LayoutWith store={store} routes={routes} />
      </Router>
    </Provider>,
  );

  it('Renders correctly default', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Can toggle drawer', () => {
    const dumbComponent = component.root.findByType(DumbLayout);
    dumbComponent.instance.toggleDrawer();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
