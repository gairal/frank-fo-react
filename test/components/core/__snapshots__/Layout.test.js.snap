// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > core > Layout Can toggle drawer 1`] = `
Array [
  <div
    className="HiddenCss-smDown-8"
  >
    <div
      className="MuiDrawer-docked-18"
    >
      <div
        className="MuiPaper-root-29 MuiPaper-elevation0-31 MuiDrawer-paper-19 SideNav-drawerPaper-2 MuiDrawer-paperAnchorLeft-20 MuiDrawer-paperAnchorDockedLeft-24"
      >
        <div>
          <header
            className="MuiPaper-root-29 MuiPaper-elevation4-35 MuiAppBar-root-56 MuiAppBar-positionStatic-60 MuiAppBar-colorPrimary-62"
          >
            <div
              className="MuiToolbar-root-64 MuiToolbar-regular-66 MuiToolbar-gutters-65"
            >
              <h1
                className="MuiTypography-root-68 MuiTypography-title-74 MuiTypography-colorInherit-87"
              >
                frank g.
              </h1>
            </div>
          </header>
          <ul
            className="MuiList-root-93 MuiList-padding-94"
          >
            <a
              aria-current="page"
              className="nav-link active"
              href="/works"
              onClick={[Function]}
              style={Object {}}
            >
              <div
                className="MuiButtonBase-root-107 MuiListItem-root-97 MuiListItem-default-100 MuiListItem-gutters-104 MuiListItem-button-105"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex="0"
              >
                <span
                  aria-hidden="true"
                  className="material-icons MuiIcon-root-111 MuiIcon-colorError-115 MuiListItemIcon-root-110"
                >
                  work
                </span>
                <div
                  className="MuiListItemText-root-118"
                >
                  <span
                    className="MuiTypography-root-68 MuiTypography-subheading-75 MuiListItemText-primary-121"
                  >
                    works
                  </span>
                </div>
                <span
                  className="MuiTouchRipple-root-124"
                />
              </div>
            </a>
            <a
              aria-current={null}
              className="nav-link"
              href="/educations"
              onClick={[Function]}
            >
              <div
                className="MuiButtonBase-root-107 MuiListItem-root-97 MuiListItem-default-100 MuiListItem-gutters-104 MuiListItem-button-105"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex="0"
              >
                <span
                  aria-hidden="true"
                  className="material-icons MuiIcon-root-111 MuiIcon-colorError-115 MuiListItemIcon-root-110"
                >
                  account_balance
                </span>
                <div
                  className="MuiListItemText-root-118"
                >
                  <span
                    className="MuiTypography-root-68 MuiTypography-subheading-75 MuiListItemText-primary-121"
                  >
                    educations
                  </span>
                </div>
                <span
                  className="MuiTouchRipple-root-124"
                />
              </div>
            </a>
            <a
              aria-current={null}
              className="nav-link"
              href="/skills"
              onClick={[Function]}
            >
              <div
                className="MuiButtonBase-root-107 MuiListItem-root-97 MuiListItem-default-100 MuiListItem-gutters-104 MuiListItem-button-105"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex="0"
              >
                <span
                  aria-hidden="true"
                  className="material-icons MuiIcon-root-111 MuiIcon-colorError-115 MuiListItemIcon-root-110"
                >
                  build
                </span>
                <div
                  className="MuiListItemText-root-118"
                >
                  <span
                    className="MuiTypography-root-68 MuiTypography-subheading-75 MuiListItemText-primary-121"
                  >
                    skills
                  </span>
                </div>
                <span
                  className="MuiTouchRipple-root-124"
                />
              </div>
            </a>
            <a
              aria-current={null}
              className="nav-link"
              href="/interests"
              onClick={[Function]}
            >
              <div
                className="MuiButtonBase-root-107 MuiListItem-root-97 MuiListItem-default-100 MuiListItem-gutters-104 MuiListItem-button-105"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex="0"
              >
                <span
                  aria-hidden="true"
                  className="material-icons MuiIcon-root-111 MuiIcon-colorError-115 MuiListItemIcon-root-110"
                >
                  language
                </span>
                <div
                  className="MuiListItemText-root-118"
                >
                  <span
                    className="MuiTypography-root-68 MuiTypography-subheading-75 MuiListItemText-primary-121"
                  >
                    interests
                  </span>
                </div>
                <span
                  className="MuiTouchRipple-root-124"
                />
              </div>
            </a>
            <hr
              className="MuiDivider-root-131"
            />
            <div
              className="MuiButtonBase-root-107 MuiListItem-root-97 MuiListItem-default-100 MuiListItem-gutters-104 MuiListItem-button-105"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex="0"
            >
              <div
                className="MuiListItemText-root-118"
              >
                <span
                  className="MuiTypography-root-68 MuiTypography-subheading-75 MuiListItemText-primary-121"
                >
                  version
                </span>
                <p
                  className="MuiTypography-root-68 MuiTypography-body1-77 MuiTypography-colorTextSecondary-91 MuiListItemText-secondary-122"
                >
                  4.2.0-react
                </p>
              </div>
              <span
                className="MuiTouchRipple-root-124"
              />
            </div>
          </ul>
        </div>
      </div>
    </div>
  </div>,
  <header
    className="MuiPaper-root-29 MuiPaper-elevation4-35 MuiAppBar-root-56 MuiAppBar-positionFixed-57 MuiAppBar-colorPrimary-62 mui-fixed Header-appBar-135"
  >
    <div
      className="MuiToolbar-root-64 MuiToolbar-regular-66 MuiToolbar-gutters-65"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root-107 MuiIconButton-root-137 MuiIconButton-colorInherit-138 Header-navIconHide-136"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-142"
        >
          <span
            aria-hidden="true"
            className="material-icons MuiIcon-root-111"
          >
            menu
          </span>
        </span>
        <span
          className="MuiTouchRipple-root-124"
        />
      </button>
      <h1
        className="MuiTypography-root-68 MuiTypography-display1-72 MuiTypography-colorInherit-87"
        style={
          Object {
            "flex": 1,
          }
        }
      >
        works
      </h1>
      <button
        aria-label="options"
        className="MuiButtonBase-root-107 MuiIconButton-root-137 MuiIconButton-colorInherit-138"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-142"
        >
          <span
            aria-hidden="true"
            className="material-icons MuiIcon-root-111"
          >
            more_vert
          </span>
        </span>
        <span
          className="MuiTouchRipple-root-124"
        />
      </button>
    </div>
  </header>,
  <div
    className="MuiLinearProgress-root-244 MuiLinearProgress-colorPrimary-245"
    mode="query"
    role="progressbar"
  >
    <div
      className="MuiLinearProgress-bar-252 MuiLinearProgress-barColorPrimary-253 MuiLinearProgress-bar1Indeterminate-255"
      style={Object {}}
    />
    <div
      className="MuiLinearProgress-bar-252 MuiLinearProgress-barColorPrimary-253 MuiLinearProgress-bar2Indeterminate-258"
      style={Object {}}
    />
  </div>,
  <main
    className="Layout-content-1"
  >
    <section>
      <div
        className="MuiGrid-container-147"
      />
    </section>
  </main>,
]
`;

exports[`components > core > Layout Renders correctly default 1`] = `
Array [
  <div
    className="HiddenCss-smDown-8"
  >
    <div
      className="MuiDrawer-docked-18"
    >
      <div
        className="MuiPaper-root-29 MuiPaper-elevation0-31 MuiDrawer-paper-19 SideNav-drawerPaper-2 MuiDrawer-paperAnchorLeft-20 MuiDrawer-paperAnchorDockedLeft-24"
      >
        <div>
          <header
            className="MuiPaper-root-29 MuiPaper-elevation4-35 MuiAppBar-root-56 MuiAppBar-positionStatic-60 MuiAppBar-colorPrimary-62"
          >
            <div
              className="MuiToolbar-root-64 MuiToolbar-regular-66 MuiToolbar-gutters-65"
            >
              <h1
                className="MuiTypography-root-68 MuiTypography-title-74 MuiTypography-colorInherit-87"
              >
                frank g.
              </h1>
            </div>
          </header>
          <ul
            className="MuiList-root-93 MuiList-padding-94"
          >
            <a
              aria-current="page"
              className="nav-link active"
              href="/works"
              onClick={[Function]}
              style={Object {}}
            >
              <div
                className="MuiButtonBase-root-107 MuiListItem-root-97 MuiListItem-default-100 MuiListItem-gutters-104 MuiListItem-button-105"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex="0"
              >
                <span
                  aria-hidden="true"
                  className="material-icons MuiIcon-root-111 MuiIcon-colorError-115 MuiListItemIcon-root-110"
                >
                  work
                </span>
                <div
                  className="MuiListItemText-root-118"
                >
                  <span
                    className="MuiTypography-root-68 MuiTypography-subheading-75 MuiListItemText-primary-121"
                  >
                    works
                  </span>
                </div>
                <span
                  className="MuiTouchRipple-root-124"
                />
              </div>
            </a>
            <a
              aria-current={null}
              className="nav-link"
              href="/educations"
              onClick={[Function]}
            >
              <div
                className="MuiButtonBase-root-107 MuiListItem-root-97 MuiListItem-default-100 MuiListItem-gutters-104 MuiListItem-button-105"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex="0"
              >
                <span
                  aria-hidden="true"
                  className="material-icons MuiIcon-root-111 MuiIcon-colorError-115 MuiListItemIcon-root-110"
                >
                  account_balance
                </span>
                <div
                  className="MuiListItemText-root-118"
                >
                  <span
                    className="MuiTypography-root-68 MuiTypography-subheading-75 MuiListItemText-primary-121"
                  >
                    educations
                  </span>
                </div>
                <span
                  className="MuiTouchRipple-root-124"
                />
              </div>
            </a>
            <a
              aria-current={null}
              className="nav-link"
              href="/skills"
              onClick={[Function]}
            >
              <div
                className="MuiButtonBase-root-107 MuiListItem-root-97 MuiListItem-default-100 MuiListItem-gutters-104 MuiListItem-button-105"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex="0"
              >
                <span
                  aria-hidden="true"
                  className="material-icons MuiIcon-root-111 MuiIcon-colorError-115 MuiListItemIcon-root-110"
                >
                  build
                </span>
                <div
                  className="MuiListItemText-root-118"
                >
                  <span
                    className="MuiTypography-root-68 MuiTypography-subheading-75 MuiListItemText-primary-121"
                  >
                    skills
                  </span>
                </div>
                <span
                  className="MuiTouchRipple-root-124"
                />
              </div>
            </a>
            <a
              aria-current={null}
              className="nav-link"
              href="/interests"
              onClick={[Function]}
            >
              <div
                className="MuiButtonBase-root-107 MuiListItem-root-97 MuiListItem-default-100 MuiListItem-gutters-104 MuiListItem-button-105"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex="0"
              >
                <span
                  aria-hidden="true"
                  className="material-icons MuiIcon-root-111 MuiIcon-colorError-115 MuiListItemIcon-root-110"
                >
                  language
                </span>
                <div
                  className="MuiListItemText-root-118"
                >
                  <span
                    className="MuiTypography-root-68 MuiTypography-subheading-75 MuiListItemText-primary-121"
                  >
                    interests
                  </span>
                </div>
                <span
                  className="MuiTouchRipple-root-124"
                />
              </div>
            </a>
            <hr
              className="MuiDivider-root-131"
            />
            <div
              className="MuiButtonBase-root-107 MuiListItem-root-97 MuiListItem-default-100 MuiListItem-gutters-104 MuiListItem-button-105"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex="0"
            >
              <div
                className="MuiListItemText-root-118"
              >
                <span
                  className="MuiTypography-root-68 MuiTypography-subheading-75 MuiListItemText-primary-121"
                >
                  version
                </span>
                <p
                  className="MuiTypography-root-68 MuiTypography-body1-77 MuiTypography-colorTextSecondary-91 MuiListItemText-secondary-122"
                >
                  4.2.0-react
                </p>
              </div>
              <span
                className="MuiTouchRipple-root-124"
              />
            </div>
          </ul>
        </div>
      </div>
    </div>
  </div>,
  <header
    className="MuiPaper-root-29 MuiPaper-elevation4-35 MuiAppBar-root-56 MuiAppBar-positionFixed-57 MuiAppBar-colorPrimary-62 mui-fixed Header-appBar-135"
  >
    <div
      className="MuiToolbar-root-64 MuiToolbar-regular-66 MuiToolbar-gutters-65"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root-107 MuiIconButton-root-137 MuiIconButton-colorInherit-138 Header-navIconHide-136"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-142"
        >
          <span
            aria-hidden="true"
            className="material-icons MuiIcon-root-111"
          >
            menu
          </span>
        </span>
        <span
          className="MuiTouchRipple-root-124"
        />
      </button>
      <h1
        className="MuiTypography-root-68 MuiTypography-display1-72 MuiTypography-colorInherit-87"
        style={
          Object {
            "flex": 1,
          }
        }
      >
        works
      </h1>
      <button
        aria-label="options"
        className="MuiButtonBase-root-107 MuiIconButton-root-137 MuiIconButton-colorInherit-138"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-142"
        >
          <span
            aria-hidden="true"
            className="material-icons MuiIcon-root-111"
          >
            more_vert
          </span>
        </span>
        <span
          className="MuiTouchRipple-root-124"
        />
      </button>
    </div>
  </header>,
  <div
    className="MuiLinearProgress-root-244 MuiLinearProgress-colorPrimary-245"
    mode="query"
    role="progressbar"
  >
    <div
      className="MuiLinearProgress-bar-252 MuiLinearProgress-barColorPrimary-253 MuiLinearProgress-bar1Indeterminate-255"
      style={Object {}}
    />
    <div
      className="MuiLinearProgress-bar-252 MuiLinearProgress-barColorPrimary-253 MuiLinearProgress-bar2Indeterminate-258"
      style={Object {}}
    />
  </div>,
  <main
    className="Layout-content-1"
  >
    <section>
      <div
        className="MuiGrid-container-147"
      />
    </section>
  </main>,
]
`;
